src/plugins/ResizeReorder/ResizerKnob.js

```diff
@@ -0,0 +1,187 @@
+
+  initialState = {
+    /**
+     * @type {boolean} Set true when column resizing starts. It is used to make ResizerLine visible.
+     */
+    isColumnResizing: undefined,
+
+    /**
+     * @type {number} X coordinate of ResizerLine during resizing. It is passed do to ResizerLine to render at appropriate position.
+     */
+    currentMouseXCoordinate: 0,
+
+    /**
+     * @type {number} Total displacement of mouse calculated from initial position when resizing started
+     */
+    totalDisplacement: 0,
+
+    /**
+     * @type {number} Top position of ResizerKnow. It is passed to ResizerLine to render at appropriate position.
+     */
+    top: 0,
+  };
+
+  state = { ...this.initialState };
+
+  // Ref to ResizerKnob
+  curRef = null;
+
+  /**
+   *
+   * @type {DOMMouseMoveTracker}
+   */
+  mouseMoveTracker = null;
+
+  constructor(props) {
+    super(props);
+  }
+
+
+  componentDidMount() {
+    this.setState({
+      top: this.curRef.getBoundingClientRect().top
+    });
+  }
+
+  render() {
+    const resizerKnobStyle = {
+      height: this.props.height
+    };
+    const resizerLine =
+      <ResizerLine
+        height={this.props.resizerLineHeight}
+        visible={!!this.state.isColumnResizing}
+        xCoordinate={this.state.currentMouseXCoordinate}
+        top={this.state.top}
+      />;
+
+    return (
+      <div
+        className={cx('fixedDataTableCellLayout/columnResizerContainer')}
+        ref={(element) => this.curRef = element}
+        style={resizerKnobStyle}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? this.suppressEvent : null}
+        onTouchMove={this.props.touchEnabled ? this.suppressEvent : null}>
+        {resizerLine}
+      </div>
+    );
+  }
+
+  /**
+   * Registers event listeners for mouse tracking
+   * @param {MouseEvent} event
+   */
+  initializeDOMMouseMoveTracker = (event) => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+    this.mouseMoveTracker.captureMouseMoves(event);
+  };
+
+  /**
+   * @param {MouseEvent} ev Mouse down event
+   */
+  onMouseDown = (ev) => {
+    this.initializeDOMMouseMoveTracker(ev);
+    const initialMouseXCoordinate = FixedDataTableEventHelper.getCoordinatesFromEvent(ev).x;
+    this.setState({
+      initialMouseXCoordinate,
+      isColumnResizing: true,
+      totalDisplacement: 0,
+      currentMouseXCoordinate: initialMouseXCoordinate
+    });
+  };
+
+  onMouseUp = () => {
+    const { minWidth, maxWidth } = this.getMinMaxWidth();
+    const newWidth = clamp(this.props.width + (this.state.totalDisplacement * (this.props.isRTL ? -1 : 1)), minWidth, maxWidth);
+    this.mouseMoveTracker.releaseMouseMoves();
+    this.resetColumnResizing(() => this.props.onColumnResizeEnd(newWidth, this.props.columnKey));
```

Comment:
First reset ResizerKnob state and then inform user about the new width of column